lib/core/src/model.rs

```diff
@@ -69,6 +72,23 @@ impl Config {
         }
     }
 
+    pub(crate) fn get_wallet_working_dir(
+        &self,
+        config: &Config,
+        mnemonic: &str,
```

Comment:
Why not pass `signer.xpub().public_key` as the arg here instead of `mnemonic`?

It would limit the number of places `mnemonic` is propagated to, yet would still give us the PK we want.

Reply:
Wouldn't it be better to use `signer.fingerprint()` instead of the xpub PK? The fingerprint is the first 4 bytes of the HASH160 of the PK.

It's shorter (4 bytes / 8 hex chars) compared to the PK (33 bytes / 66 chars).